### PR TITLE
core: some follow-ups

### PR DESCRIPTION
**Pull request by YHNdnzj**
Replaces #31901

### CODE REVIEW

File: src/core/swap.c [hunk 1]
```diff
@@ -395,51 +394,42 @@ static int swap_setup_unit(
...
-                r = unit_new_for_name(m, sizeof(Swap), e, &u);
-                if (r < 0) {
-                        log_unit_warning_errno(u, r, "Failed to load swap unit: %m");
-                        goto fail;
-                }
+                r = unit_new_for_name(m, sizeof(Swap), e, &new);
+                if (r < 0)
+                        return log_warning_errno(r, "Failed to load swap unit '%s': %m", e);
 
+                u = new;
```
Comment (daandemeyer) [inline src/core/swap.c:405]: Previously this was not cleaned up, now it is? Is that intentional?

**Review by DaanDeMeyer — CHANGES_REQUESTED** [review]

File: src/core/swap.c (hunk 1 above)
Comment (YHNdnzj) [reply src/core/swap.c:405]: That was also a bug introduced by e9fa1bf704ad2f0a7e257e29889315118b0df459 :/

**Review by DaanDeMeyer — APPROVED** [review]

File: src/core/swap.c [hunk 2]
```diff
@@ -395,51 +394,42 @@ static int swap_setup_unit(
...
                 s = ASSERT_PTR(SWAP(u));
 
                 s->what = strdup(what);
-                if (s->what) {
-                        r = log_oom();
-                        goto fail;
-                }
+                if (!s->what)
```
Comment (yuwata) [inline src/core/swap.c:409]: Ouch!

**Review by yuwata — APPROVED** [review]

**Review by bluca — APPROVED** [review]

File: src/core/swap.c [hunk 3]
```diff
@@ -374,11 +374,10 @@ static int swap_setup_unit(
                 int priority,
                 bool set_flags) {
 
+        _cleanup_(unit_freep) Unit *new = NULL;
```
Comment (poettering) [inline src/core/swap.c:377]: new() is a macro. It's a bit weird naming a variable like that.

Comment (poettering) [reply src/core/swap.c:377]: (yeah, i know pre-existing issue, but still, if you touch it...)

File: src/core/swap.c (hunk 2 above)
Comment (poettering) [reply src/core/swap.c:409]: uff

File: src/core/swap.c (hunk 3 above)
Comment (poettering) [inline src/core/swap.c:377]: hmm, this variable always goes out of scope and is destroyed. There's some TAKE_PTR(new) missing below in the succcess path?

Comment (poettering) [reply src/core/swap.c:377]: ahh, this part already got fixed in abef0ec308dc2adf90798733902bff23a7e3a916. Ignore this comment hence.

**Pull request merged**